runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -308,11 +310,14 @@ end
 --- )
 --- ```
 ---
+---@param result lsp.DocumentDiagnosticReport
```

Comment:
Isn't this `DocumentDiagnosticParams`? See https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_publishDiagnostics

Reply:
Oh, I confused it with `on_publish_diagnostics`. Nevermind